src/EnergyPlus/UnitarySystem.cc

```diff
@@ -525,56 +449,92 @@ namespace UnitarySystems {
+                        // not sure if new cooling coil model needs to set something? but certainly can't call DXCoils
+                    } else {
+                        DXCoils::SetCoilSystemCoolingData(state, this->m_CoolingCoilName, this->Name);
+                    }
                 }
-                // open this up to include UnitarySystem on a future commit
-                if (!this->m_FanExists && this->m_sysType == SysType::CoilCoolingDX) {
-                    if (this->m_CoolCoilExists) { // copy this block for heating coil also on future commit
-                        bool coilDataSet = false;
-                        bool errorsFound = false;
-                        std::string coilName;
-                        std::string coilType;
-                        if (this->m_CoolingCoilType_Num == DataHVACGlobals::Coil_CoolingAirToAirVariableSpeed) {
-                            VariableSpeedCoils::getCoilTypeAndName(state, this->m_CoolingCoilIndex, coilType, coilName, errorsFound);
-                            if (!errorsFound) coilDataSet = true;
-                        } else if (this->m_CoolingCoilIndex == DataHVACGlobals::CoilDX_CoolingSingleSpeed) {
-                            // need to check for all other coil types
+                // associates an air loop fan on main branch with a coil on main branch where parent does not have a fan
+                if (!this->m_FanExists) {
+                    if (state.dataAirSystemsData->PrimaryAirSystems(AirLoopNum).supFanModelType != DataAirSystems::Invalid) {
+                        if (this->m_CoolCoilExists) {
+                            state.dataRptCoilSelection->coilSelectionReportObj->setCoilSupplyFanInfo(
+                                state,
+                                this->m_CoolingCoilName,
+                                DataHVACGlobals::cAllCoilTypes(this->m_CoolingCoilIndex),
+                                this->m_FanType_Num == DataHVACGlobals::FanType_SystemModelObject
+                                    ? state.dataHVACFan->fanObjs[state.dataAirSystemsData->PrimaryAirSystems(AirLoopNum).supFanVecIndex]->name
+                                    : state.dataFans->Fan(state.dataAirSystemsData->PrimaryAirSystems(AirLoopNum).SupFanNum).FanName,
+                                this->m_FanType_Num == DataHVACGlobals::FanType_SystemModelObject ? DataAirSystems::ObjectVectorOOFanSystemModel
+                                                                                                  : DataAirSystems::StructArrayLegacyFanModels,
+                                this->m_FanType_Num == DataHVACGlobals::FanType_SystemModelObject
+                                    ? state.dataAirSystemsData->PrimaryAirSystems(AirLoopNum).supFanVecIndex
+                                    : state.dataAirSystemsData->PrimaryAirSystems(AirLoopNum).SupFanNum);
                         }
```

Comment:
Not sure if this warning is an issue.

![image](https://user-images.githubusercontent.com/5778156/166158384-1715e619-aeab-4734-a022-bd73938cad03.png)